src/unix/apple/disk.rs

```diff
@@ -72,44 +72,78 @@ impl DiskInner {
         self.is_read_only
     }
 
-    pub(crate) fn refresh(&mut self) -> bool {
-        #[cfg(target_os = "macos")]
-        let Some((read_bytes, written_bytes)) = self
-            .bsd_name
-            .as_ref()
-            .and_then(|name| crate::sys::inner::disk::get_disk_io(name))
-        else {
-            sysinfo_debug!("Failed to update disk i/o stats");
-            return false;
+    pub(crate) fn refresh_specifics(&mut self, refresh_kind: DiskRefreshKind) -> bool {
+        let type_ = if refresh_kind.kind() {
```

Comment:
Same here: please only update the kind if `refresh_kind.kind()` or if it's `DiskKind::Unknown`.